clients/spark-extensions/pom.xml

```diff
@@ -33,7 +33,7 @@
     <dependency>
       <groupId>org.projectnessie</groupId>
       <artifactId>nessie-client</artifactId>
-      <version>${project.version}</version>
+      <version>${iceberg.nessie.version}</version>
```

Comment:
Does this particular one really need to be bumped?
I'm a bit worried about this particular one, because it's a production-dependency and `nessie-client` depends on `nessie-model`.

Reply:
yes, this definitely needs to be bumped

Reply:
you mean we should also exclude the model classes here?

Reply:
I had another look at the pom and turns out this was running with the nessie-quarkus `project.version` and not the `iceberg.nessie.version`. Pushed a fix for this and also forced the `nessie-model` version to be `iceberg.nessie.version`

Reply:
Really not happy w/ having these versions as _production_ dependencies here. I think, that makes it really complicated to reason about and prone to more difficult dependency issues.

It's already unclear, which nessie and 3rd party dependencies will land here in which versions. `nessie-client-tests` brings in more dependencies.

The changes to the gc-modules are fine IMO, because those only affect tests, but this change affects production dependencies.

Reply:
@rymurr thoughts on @snazy's concerns? I currently don't see any easier way around this problem than running the spark sql extensions code with the nessie version that comes with iceberg.
imo we need to rethink the dependencies between nessie and iceberg entirely as the way it is currently structured is just error prone

Reply:
why can't this just be `provided` or `optional` scope? We just don't ship it with the shaded jar or as a dependency and then the delta or iceberg runtimes use their nessie client that is already available to them.

I thikn we need to make sure it generates a dependency reduced pom and that shading is correct though

Reply:
fixed